enterprise/frontend/src/embedding/data-picker/DataSelector/tests/DataSourceSelector.unit.spec.tsx

```diff
@@ -0,0 +1,185 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import { getNextId } from "__support__/utils";
+import type { Database, Table } from "metabase-types/api";
+import { createMockDatabase, createMockTable } from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { DataSourceSelector } from "../DataSelector";
+
+const DATABASES = [createSampleDatabase()];
+
+const storeInitialState = createMockState({
+  settings: createMockSettingsState({
+    "enable-nested-queries": true,
+  }),
+});
+
+const AVAILABLE_MODELS: Record<AvailableModels, ("dataset" | "table")[]> = {
+  "tables-only": ["table"],
+  "tables-and-models": ["dataset", "table"],
+};
+
+type AvailableModels = "tables-only" | "tables-and-models";
+
+interface SetupOpts {
+  databases?: Database[];
+  isJoinStep?: boolean;
+  availableModels?: AvailableModels;
+}
+
+function setup({
+  databases = DATABASES,
+  isJoinStep = false,
+  availableModels = "tables-only",
+}: SetupOpts = {}) {
+  fetchMock.get(
+    {
+      url: "path:/api/search",
+      query: {
+        calculate_available_models: true,
+        limit: 0,
+        models: ["dataset"],
+      },
+    },
+    {
+      data: [],
+      limit: 0,
+      models: ["dataset"],
+      offset: 0,
+      table_db_id: null,
+      engine: "search.engine/in-place",
+      total: 1,
+      available_models: AVAILABLE_MODELS[availableModels],
+    },
+  );
+
+  setupDatabasesEndpoints(databases, undefined, { saved: true });
+
+  return renderWithProviders(
+    <DataSourceSelector
+      isInitiallyOpen
+      canChangeDatabase={!isJoinStep}
+      selectedDatabaseId={null}
+      canSelectModel={true}
+      canSelectTable={true}
+      triggerElement={<div>Click me to open or close data picker</div>}
+      setSourceTableFn={jest.fn()}
+    />,
+    { storeInitialState },
+  );
+}
+
+describe("DataSourceSelector", () => {
```

Comment:
This file basically tests the [main use cases defined here](https://www.notion.so/metabase/Test-Extract-embedding-picker-out-of-the-current-data-picker-1e469354c90180379ea1c3160cfe56b7?pvs=4#1e469354c90180b39bffc0d339c4b04f)